2015/introducing-snabbwall.markdown

```diff
@@ -0,0 +1,95 @@
+Title: Introducing SnabbWall
+Date: 2015-12-31 20:00:00
+Author: aperez
+Tags: igalia, snabb, snabbwall
+
+If you have been following the blogs of my colleagues [Katerina](http://luatime.org), [Diego](http://blogs.igalia.com/dpino/), and [Andy](http://wingolog.org) it should not be a suprise for you to know that in [Igalia](http://www.igalia.com) we have a small —and awesome!— team working with [Snabb Switch](http://snabb.co) to provide novel software–mostly solutions for networking.
+
+We are now starting the development of [SnabbWall](http://snabbwall.org) is an *application-level* (Layer-7) firewall *suite* for [Snabb Switch](http://snabb.co), and yours truly happens to be the lead developer, so I figured out this is a good moment to introduce the project, which is kindly sponsored by the [NLnet Foundation](http://nlnet.nl/).
+
+
+<figure style="text-align:center">
+  ![](http://snabbwall.org/images/igalia-logo.png)
+  <span style="margin-left:20px">&nbsp;</span>
+  ![](http://snabbwall.org/images/nlnet-logo.gif)
+</figure>
+
+The one-minute introduction to Snabb
+------------------------------------
+
+Snabb works by taking over network interfaces, bypassing the kernel drivers, and using its own. When used to implement a certain network functinoality, what your (Snabb) program “sees” is no more than streams of raw packets. Gone are the TCP/IP stack, BSD sockets, and all the other functionality provided by the operating system kernel to user space programs. Heck, you even need to roll your [own](https://github.com/SnabbCo/snabbswitch/tree/master/src/apps/intel) [drivers](https://github.com/SnabbCo/snabbswitch/tree/master/src/apps/solarflare)! There is nothing in there but the hardware waiting to have streams of electrons blasted through it. But things go *real* fast in there without the overhead of a traditional network stack, so despite the void it is a good place to massage packets and pass them along.
```

Comment:
Typo fixed, and also fixed “suprise” in the first paragraph.

I mean “massage packets”, as written. As in “a packet comes, your code gives it a massage *(meaning: ‘manipulates it’)*, and then it is sent somewhere else”.